web/components/contract/contract-tabs.tsx

```diff
@@ -139,95 +137,27 @@ const CommentsTabContent = memo(function CommentsTabContent(props: {
-            </Row>
-          </Tooltip>
-        </button>
-      </Row>
-    </Row>
-  )
-  if (contract.outcomeType === 'FREE_RESPONSE') {
-    return (
-      <>
-        <ContractCommentInput className="mb-5" contract={contract} />
-        {sortRow}
-        {answerResponse && (
-          <AnswerCommentInput
-            contract={contract}
-            answerResponse={answerResponse}
-            onCancelAnswerResponse={onCancelAnswerResponse}
-          />
-        )}
-        {topLevelComments.map((parent) => {
-          if (parent.answerOutcome === undefined) {
-            return (
-              <FeedCommentThread
-                key={parent.id}
-                contract={contract}
-                parentComment={parent}
-                threadComments={sortBy(
-                  commentsByParent[parent.id] ?? [],
-                  (c) => c.createdTime
-                )}
-                tips={tips}
-              />
-            )
-          }
-          const answer = contract.answers.find(
-            (answer) => answer.id === parent.answerOutcome
-          )
-          if (answer === undefined) {
-            console.error('Could not find answer that matches ID')
-            return <></>
-          }
-          return (
-            <>
-              <Row className="gap-2">
-                <CommentsAnswer answer={answer} contract={contract} />
-              </Row>
-              <Row>
-                <div className="ml-1">
-                  <Curve size={28} strokeWidth={1} color="#D8D8EB" />
-                </div>
-                <div className="w-full pt-1">
-                  <FeedCommentThread
-                    key={parent.id}
-                    contract={contract}
-                    parentComment={parent}
-                    threadComments={sortBy(
-                      commentsByParent[parent.id] ?? [],
-                      (c) => c.createdTime
-                    )}
-                    tips={tips}
-                  />
-                </div>
-              </Row>
-            </>
-          )
-        })}
-      </>
-    )
-  } else {
-    return (
-      <>
-        <ContractCommentInput className="mb-5" contract={contract} />
-        {sortRow}
-
-        {topLevelComments.map((parent) => (
+  return (
+    <>
+      <ContractCommentInput className="mb-5" contract={contract} />
+      <SortRow
+        comments={comments}
+        contract={contract}
+        sort={sort}
+        onSortClick={() => setSort(sort === 'Newest' ? 'Best' : 'Newest')}
+      />
+      {contract.outcomeType === 'FREE_RESPONSE' && (
+        <FreeResponseComments
+          contract={contract}
+          answerResponse={answerResponse}
+          onCancelAnswerResponse={onCancelAnswerResponse}
+          topLevelComments={topLevelComments}
+          commentsByParent={commentsByParent}
+          tips={tips}
+        />
+      )}
+      {contract.outcomeType != 'FREE_RESPONSE' &&
```

Comment:
If you're using `===` above, might as well use `!==` here. I think it's good to default to the extra equals anyway.

The only differences are roughly `null == undefined` `'1' == 1` evaluate true, but evaluate false with `===`.